### PR TITLE
[MM-46218] Set focus to highlighted post

### PR DESCRIPTION
**Pull request by kostaspt**
#### Summary
When there is a highlighted message, shift+tab is expected to navigate there (when using a screen reader). Currently, no matter what, the last message is always highlighted.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46218

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/post/post_component.tsx [hunk 1]
```diff
@@ -130,24 +130,43 @@ const PostComponent = (props: Props): JSX.Element => {
...
         return undefined;
-    }, [props.shouldHighlight]);
+    }, [shouldHighlight]);
 
     const handleA11yActivateEvent = () => setA11y(true);
     const handleA11yDeactivateEvent = () => setA11y(false);
     const handleAlt = (e: KeyboardEvent) => setAlt(e.altKey);
 
+    const handleA11yKeyboardFocus = useCallback((e: KeyboardEvent) => {
+        if (e.key === Constants.KeyCodes.TAB[0] && e.shiftKey && shouldHighlight && !hasReceivedA11yFocus) {
```
Comment (AshishDhama) [inline webapp/channels/src/components/post/post_component.tsx:153]: Nit: But may be instead we can use https://github.com/mattermost/mattermost-server/blob/a85a7c84220ee9f9c83d729fa5dbfb24abd01b20/webapp/channels/src/utils/keyboard.ts#L16

**Review by AshishDhama — APPROVED** [review]

File: webapp/channels/src/components/post/post_component.tsx (hunk 1 above)
Comment (kostaspt) [reply webapp/channels/src/components/post/post_component.tsx]: Didn't know this existed, thank you!

**Pull request merged**